src/main/java/com/box/sdk/BoxResourceIterable.java

```diff
@@ -0,0 +1,158 @@
+package com.box.sdk;
+
+import java.net.URL;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+import com.eclipsesource.json.JsonArray;
+import com.eclipsesource.json.JsonObject;
+
+/**
+ * Common implementation for paging support.
+ *
+ * @author Stanislav Dvorscak
+ *
+ * @param <T>
+ *            type of iterated entity
+ */
+public abstract class BoxResourceIterable<T> implements Iterable<T> {
```

Comment:
According to the Web Hook documentation, Web Hooks paging is based on "next", but JSONIterator is based on "limit" and "offset". Because of this, I have created new implementation and I think paging should be only one, maybe JSONIterator can be deprecated, because limit and offset is not efficient way for paging. Other reason is generic implementation, which avoid duplicate code: isNext. And in case of moving forward to Java 8, it will be more easy for implementation.